flow/core/rewards.py

```diff
@@ -330,3 +330,93 @@ def energy_consumption(env, gain=.001):
         power += M * speed * accel + M * g * Cr * speed + 0.5 * rho * A * Ca * speed ** 3
 
     return -gain * power
+
+
+def veh_energy_consumption(env, veh_id, gain=.001):
+    """Calculate power consumption of a vehicle.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the actual power consumed
+    by a vehicle.
+    """
+    power = 0
+
+    M = 1200  # mass of average sized vehicle (kg)
+    g = 9.81  # gravitational acceleration (m/s^2)
+    Cr = 0.005  # rolling resistance coefficient
+    Ca = 0.3  # aerodynamic drag coefficient
+    rho = 1.225  # air density (kg/m^3)
+    A = 2.6  # vehicle cross sectional area (m^2)
+    speed = env.k.vehicle.get_speed(veh_id)
+    prev_speed = env.k.vehicle.get_previous_speed(veh_id)
+
+    accel = abs(speed - prev_speed) / env.sim_step
+
+    power += M * speed * accel + M * g * Cr * speed + 0.5 * rho * A * Ca * speed ** 3
+
+    return -gain * power
+
+
+def miles_per_megajoule(env, veh_id=None, gain=.001):
+    """Calculate miles per mega-joule of either a particular vehicle or the total average of all the vehilces.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the actual power consumed
+    by a vehicle.
+    """
+    mpj = 0
+    counter = 0
+    if not isinstance(veh_id, list):
+        speed = env.k.vehicle.get_speed(veh_id)
+        # convert to be positive since the function called is a penalty
+        power = -veh_energy_consumption(env, veh_id, gain=1.0)
+        if power > 0 and speed >= 0.0:
+            # meters / joule is (v * \delta t) / (power * \delta t)
+            mpj = speed / power
+    else:
+        for veh_id in env.k.vehicle.get_ids():
```

Comment:
shouldn't you iterate over vehicle.get_ids() if veh_id is None? here veh_id can be a list, shouldn't you iterate over veh_id list?